WooCommerce/Classes/ViewRelated/Orders/Order Filters/Date Range Filter/Custom Range Filter/DateRangeFilterViewController.swift

```diff
@@ -0,0 +1,231 @@
+import UIKit
+
+// MARK: - DateRangeFilterViewController
+// Allow to filter the orders by a range of dates
+//
+final class DateRangeFilterViewController: UIViewController {
+
+    @IBOutlet private weak var tableView: UITableView!
+
+    // Completion callback
+    //
+    typealias Completion = (Date?, Date?) -> Void
+    private let onCompletion: Completion
+
+    private var startDate: Date?
+    private var endDate: Date?
+
+    // Indicate when the start or the end date picker is expanded
+    //
+    private var startDateExpanded: Bool = false
+    private var endDateExpanded: Bool = false
+
+    private var rows: [Row] = []
+
+    /// Init
+    ///
+    init(startDate: Date?,
+         endDate: Date?,
+         completion: @escaping Completion) {
+        self.startDate = startDate
+        self.endDate = endDate
+        onCompletion = completion
+        super.init(nibName: nil, bundle: nil)
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        configureNavigationBar()
+        configureMainView()
+        configureRows()
+        configureTableView()
+    }
+}
+
+// MARK: - View Configuration
+//
+private extension DateRangeFilterViewController {
+
+    func configureNavigationBar() {
+        title = Localization.navigationBarTitle
+    }
+
+    func configureMainView() {
+        view.backgroundColor = .listBackground
+    }
+
+    func configureTableView() {
+        tableView.dataSource = self
+        tableView.delegate = self
+
+        /// Registers all of the available TableViewCells
+        ///
+        for row in Row.allCases {
+            tableView.registerNib(for: row.type)
+        }
+
+        tableView.backgroundColor = .listBackground
+        tableView.removeLastCellSeparator()
+    }
+
+    func configureRows() {
```

Comment:
nit: `configure*` sounds more like a one-time setup, maybe `update*` would be more clear?